src/test-helper.ts

```diff
@@ -0,0 +1,67 @@
+import {mockTimeSource} from './mock-time-source';
+
+function makeTestHelper (options: Object = {}) {
+  return function withTime (test, testOptions: Object = {}) {
+    return function runTestWithTime (done) {
+      const Time = mockTimeSource({...options, ...testOptions});
+
+      test(Time);
+
+      // It's hard to differentiate between tape and ava
+      const usingAva = !!done && typeof done.regex === 'function';
```

Comment:
I would simply make two `makeTestHelper` functions, one `makePromiseHelper` and one `makeCallbackHelper`. Less magic, less code, works always